server/container/guice/cassandra-guice/src/test/java/org/apache/james/CassandraConfTest.java

```diff
@@ -0,0 +1,137 @@
+import org.junit.Rule;
+import org.junit.Test;
+import org.testcontainers.DockerClientFactory;
+
+public class CassandraConfTest {
+
+    private static final int IMAP_PORT = 1143;
+
+    private static String getConfStringForNodeIpAndPort(CassandraNodeIpAndPort node) {
+        if (CassandraNodeIpAndPort.DEFAULT_CASSANDRA_PORT != node.getPort()) {
+            return node.getIp() + ":" + node.getPort();
+        } else {
+            return node.getIp();
+        }
+    }
+
+    private static String getDockerHostIp() {
+        return DockerClientFactory.instance().dockerHostIpAddress();
+    }
+
+    private final DockerCassandraRule dockerCassandraRule = new DockerCassandraRule();
+
+    @Rule
+    public CassandraJmapTestRule cassandraJmapTestRule = new CassandraJmapTestRule(dockerCassandraRule,
+            new EmbeddedElasticSearchRule());
+
+    private GuiceJamesServerImpl jamesServer;
+    private SocketChannel socketChannel;
+
+    @Before
+    public void setUp() throws IOException {
+        socketChannel = SocketChannel.open();
+    }
+
+    @After
+    public void after() throws IOException {
+        socketChannel.close();
+        if (jamesServer != null) {
+            jamesServer.stop();
+        }
+    }
+
+    @Test
+    public void serverShouldStartServiceWhenNodeIsReachable() throws Exception {
+        jamesServer = cassandraJmapTestRule.jmapServer();
+
+        assertThatServerStartCorrectly();
+    }
+
+    @Test
+    public void serverShouldStartWhenOneCassandraNodeIsUnreachable() throws Exception {
+        CassandraNodeIpAndPort unreachableNode = new CassandraNodeIpAndPort("10.2.3.42");
+
+        PropertiesConfiguration configuration = getCassandraConfigurationForDocker(unreachableNode,
+                new CassandraNodeIpAndPort(dockerCassandraRule.getIp()));
+
+        jamesServer = cassandraJmapTestRule.jmapServer(
+                (binder) -> binder.bind(CassandraSessionConfiguration.class).toInstance(() -> configuration));
+
+        assertThatServerStartCorrectly();
+    }
+
+    @Test
+    public void configShouldWorkWithNonDefaultPort() throws Exception {
+        CassandraNodeIpAndPort node = new CassandraNodeIpAndPort(getDockerHostIp(), dockerCassandraRule.getBindingPort());
+        PropertiesConfiguration configuration = getCassandraConfigurationForDocker(node);
+        jamesServer = cassandraJmapTestRule.jmapServer(
+                (binder) -> binder.bind(CassandraSessionConfiguration.class).toInstance(() -> configuration));
+
+        assertThatServerStartCorrectly();
+    }
+
+    private void assertThatServerStartCorrectly() throws Exception, IOException {
+        jamesServer.start();
+        socketChannel.connect(new InetSocketAddress("127.0.0.1", IMAP_PORT));
+        assertThat(getServerConnectionResponse(socketChannel)).startsWith("* OK JAMES IMAP4rev1 Server");
+    }
+
+    private PropertiesConfiguration getCassandraConfigurationForDocker(CassandraNodeIpAndPort... ips) {
```

Comment:
If there is one last thing I would ask you, it would be to use a String in here as argument.

You will then avoid the map reduce stuff down there (logic in test, you know) as well as allow me to see by my eyes a concrete example of such configs, even if I am not good at java.

CassandraConfTest::getConfStringForNodeIpAndPort would also become uneeded...